### PR TITLE
feat: add a provider column to the mods page

### LINKED ISSUE #402: Show source in mod list
### Role

I use Prism

### Suggestion

The mod list view could have another column for source - CurseForge, Modrinth or unknown  

### Benefit

So I can know if it was found in any platform and whether it can auto-update

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by leia-uwu**
closes #402

### CODE REVIEW

File: launcher/minecraft/mod/Mod.cpp [hunk 1]
```diff
@@ -189,4 +195,12 @@ void Mod::finishResolvingWithDetails(ModDetails&& details)
...
     if (metadata)
         setMetadata(std::move(metadata));
+};
+
+auto Mod::provider() const -> QString
+{
+    if (metadata()) {
+        return ProviderCaps.readableName(metadata()->provider);
+    }
+    return "Unknown";
```
Comment (flowln) [inline launcher/minecraft/mod/Mod.cpp]:
```suggestion
	//: Unknown mod provider (i.e. not Modrinth, CurseForge, etc...)
    return tr("Unknown");
```

**Review by flowln — COMMENTED** [review]
this is nice. however, i think it would make sense to hide the column entirely in the case where no mods have metadata (probably most cases, considering most people play modpacks and don't use that update feature). there's not much need for a column full of 'Unknown's if you aren't even going to use that :p

**Comment by flowln** [comment]
@leo78913 see if you're fine with the changes i've made, please :slightly_smiling_face: (~when i fix the build :P~ fixed)

**Comment by leia-uwu** [comment]
iam fine, thanks :D

**Review by Scrumplex — APPROVED** [review]

**Review by flowln — APPROVED** [review]
thanks!

**Pull request merged**